packages/vertexai/src/requests/response-helpers.ts

```diff
@@ -41,17 +42,19 @@ export function addHelpers(
         );
       }
       if (hadBadFinishReason(response.candidates[0])) {
-        throw ERROR_FACTORY.create(VertexError.RESPONSE_ERROR, {
-          message: `${formatBlockErrorMessage(response)}`,
-          response
-        });
+        throw new VertexAIError(
+          VertexAIErrorCode.REQUEST_ERROR,
+          `Response error: ${formatBlockErrorMessage(
+            response
+          )}. Response body stored in error.customData.response`
```

Comment:
This isn't true anymore, and it seems we're actually not storing the response anymore. It used to get stored automatically by being added as an arg to ERROR_FACTORY.create (it adds everything in the second object to customData). I guess we can put it on the top level, but you need to add an extra optional `response` property to the VertexAIError class, and then accept it in the constructor?

Reply:
Added an optional `generateContentResponse` property 👍